primitives/substrate/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Primitives for the Substrate light client (a.k.a bridge) pallet.
```

Comment:
Imo if it is primitives of the pallet, then it should have types that are used in pallet APIs. Here you have e.g. `struct ImportedHeader`, which is the struct describing how pallet is storing the header. Why to expose that? The same about other types - imo this file should be moved to smth like `pallet/storage.rs`. No pressure about that, though